examples/Vereya/4_mine_and_craft.py

```diff
@@ -0,0 +1,371 @@
+import logging
+import math
+from time import sleep
+from tagilmo.utils.malmo_wrapper import MalmoConnector, RobustObserver
+import tagilmo.utils.mission_builder as mb
+
+from tagilmo.utils.mathutils import normAngle, degree2rad
+
+import numpy as np
+import minecraft_data
+
+
+# This script shows a relatively complex behavior of gathering resources
+# for an iron pickaxe. It includes searchng for logs, mining stones, etc.
+# There are many heurisics and many things that can go wrong,
+# so the final result will not be achieved. And this is OK.
+#
+# The example is intented to show the capabilities and limitations of
+# imperative scripts and hand-coding. Failer can be due to the lack
+# of robustness and flexibility of:
+# * local skills (e.g. the agent can start mining a shaft in the direction of lava)
+# * high-level plan (the agent can run out of sticks, when the plan assumes it has enough)
+# The longer the plan, the more things can go wrong.
+# It is instructive to examine failure cases.
+
+# apparantely logs naming is different in different versions of Minecraft,
+# so we need to get right names for our current version
+
+log_names = []
+versions = minecraft_data.common().protocolVersions
+mcd = minecraft_data('1.18.1')  # here we must put current minecraft version
+for item in mcd.items_list:
+    iname = item['name']
+    if 'log' in iname:
+        log_names.append(iname)
+
+# ============== some hand-coded skills ==============
+
+# This function is executed to run to a visible object,
+# but it doesn't check if the path is safe
+def runStraight(rob, dist, keepHeight=False):
+    logging.info("\tinside runStraight")
+    start = rob.waitNotNoneObserve('getAgentPos')
+    rob.sendCommand("move 1")
+    bJump = False
+    for t in range(2 + int(dist * 5)):
+        sleep(0.1)
+        rob.observeProcCached()
+        pos = rob.getCachedObserve('getAgentPos')
+        if pos is None:
+            continue
+        if dist * dist < (pos[0] - start[0]) * (pos[0] - start[0]) + (pos[2] - start[2]) * (pos[2] - start[2]):
+            break
+        if pos[1] < start[1] - 0.5 and keepHeight:
+            rob.sendCommand("jump 1")
+            rob.sendCommand("attack 1")
+            bJump = True
+        elif bJump:
+            rob.sendCommand("jump 0")
+            rob.sendCommand("attack 0")
+            bJump = False
+        los = rob.getCachedObserve('getLineOfSights')
+        if los is not None and los['distance'] < 0.5 and \
+               not los['distance'] in RobustObserver.passableBlocks and \
```

Comment:
`not los['distance'] in RobustObserver.passableBlocks` what is checked here?

Reply:
los['distance'] is a floating point number, no?

Reply:
Ah, I see your point. It's very strange. It seems like this bug lasts even from the first Alexey's commits... Ironically it hasn't affected performance of the script yet. @DaddyWesker We can discuss how to fix this.

Reply:
`not los['distance'] in RobustObserver.passableBlocks` is always True

Reply:
@noskill  we've decided to remove this line. I've commited new changes.